logstash-core/lib/logstash/instrument/snapshot.rb

```diff
@@ -14,8 +14,8 @@ def initialize
     end
 
     def push(*args)
-      type, key, _ = args
-      metric = @metrics.fetch_or_store(key, concrete_class(type, key))
+      namespace, key, type, _ = args
+      metric = @metrics.fetch_or_store([namespace, key].join('-'), concrete_class(type, key))
```

Comment:
might be, or also push this call upstream somehow, not sure. I see this as a precomputed value as much as possible.